ambry-network/src/main/java/com.github.ambry.network/Transmission.java

```diff
@@ -166,6 +176,9 @@ public boolean hasReceive() {
   }
 
   public void clearReceive() {
+    if (networkReceive != null) {
+      ReferenceCountUtil.release(networkReceive.getReceivedBytes().getAndRelease());
```

Comment:
would this cause exceptions upon calling `getAndRelease` in `NetworkClient`? I think clearReceive is called from within the selector and then the receive is unboxed again in`NetworkClient:sendAndPoll`

Reply:
you are right, I did a test in the morning and it does mess up the network layer. Update is on the way.